### PR TITLE
feat(ld-context-menu): add context menu component

### LINKED ISSUE #438: Context menu
A context menu (also called contextual menu or pop-up menu) is a menu that appears upon user interaction, such as a right-click or a long press (in touch screen environments). It offers a limited set of choices that are available in the current state, or context, of the application to which the menu belongs. Usually the available choices are actions related to the selected object.

Some screenshots for the usage of a context menu within a task management application (first is with mouse, second with touch screen):

Screen Shot 2022-10-28 at 10 22 09
Screen Shot 2022-10-28 at 10 22 58

### PR DESCRIPTION
**Pull request by renet**
# Description

Adds a new `ld-context-menu` component, as well as additional modes for the `ld-button` component and necessary changes to the `ld-tooltip` component.

BREAKING CHANGE: danger-secondary now requires secondary class, too.

Fixes #438

## Type of change

Please delete options that are not relevant.

- [x] Feature
- [x] Bugfix
- [x] Refactoring (no functional changes, no api changes)
- [x] Documentation content changes

## Is it a breaking change?

- [x] Yes
- [ ] No

# How Has This Been Tested?

Please describe the tests that you've added and run to verify your changes. 
Provide instructions, so we can reproduce.

- [x] unit tests
- [x] e2e tests
- [x] accessibility tests
- [x] tested manually

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [x] New and existing tests pass locally with my changes

### CODE REVIEW

**Review by borisdiakur — CHANGES_REQUESTED** [review]
I haven't checked the code yet. But there are two things that I found when testing the component on preview:

1. When I focus a context menu item and press the Tab button, I land in Nirvana. I recommend preventing the default in this case.
2. The screen reader currently doesn't give any hint on focus of a context menu trigger button. Please add `aria-haspopup="menu"` to the trigger button programmatically.

It would be great if you could address these two points before I dive deeper into the code review.

**Review by borisdiakur — CHANGES_REQUESTED** [review]

File: src/liquid/components/ld-tooltip/ld-tooltip.tsx [hunk 1]
```diff
@@ -271,20 +377,21 @@ export class LdTooltip {
     return (
       <Host>
         <TriggerTag
-          aria-describedby={this.idDescriber}
+          aria-describedby={
```
[the remark this answers is not in the document]
Comment (renet) [reply src/liquid/components/ld-tooltip/ld-tooltip.tsx:380]: Will postpone this to a separate PR.

File: src/liquid/components/ld-context-menu/ld-menu/ld-menu.tsx [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+      index === 0
+        ? allMenuItems[allMenuItems.length - 1]
+        : allMenuItems[index - 1]
+
+    target.ldTabindex = -1
+    prev.ldTabindex = 0
+    prev.focusInner()
+  }
+
+  private focusByFirstCharacter = (event: KeyboardEvent) => {
```
[the remark this answers is not in the document]
Comment (renet) [reply src/liquid/components/ld-context-menu/ld-menu/ld-menu.tsx]: I finished unifying and centralizing the typeahead feature for both select and context menu components. There is one intended difference, though: in the context menu, a new search starts at the item following the currently focused item. this way I can find/focus all menu items that match a certain character or string, not only the first one. Please take a look at the implementation - there also were some "bugs" in the select component that differed from the native select behavior that have been fixed during the refactoring.

File: src/liquid/components/ld-context-menu/ld-menu/readme.md [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+
+The `ld-menu` component is used internally for the `ld-context-menu`. It is the visual part of the actual menu without the floating/tooltip functionality.
+
+## Examples
+
+### Basic
+
+{% example %}
+<ld-menu>
+  <ld-menuitem>Menum item 1</ld-menuitem>
```
Comment (borisdiakur) [inline src/liquid/components/ld-context-menu/ld-menu/readme.md:26]:
```suggestion
  <ld-menuitem>Menu item 1</ld-menuitem>
```

File: src/liquid/utils/keyboard-navigation.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+    // find the first value that is alphabetically following the query
+    const matchingLabel = [...values].sort().find((value) => query < value)
+
+    if (matchingLabel) {
+      return this._options[values.indexOf(matchingLabel)]
+    }
+  }
+
+  /* Clear timeout */
```
Comment (borisdiakur) [inline src/liquid/utils/keyboard-navigation.ts:50]:
```suggestion
```

**Review by borisdiakur — CHANGES_REQUESTED** [review]

**Review by borisdiakur — APPROVED** [review]

**Pull request merged**